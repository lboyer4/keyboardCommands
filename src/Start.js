import React, { Component } from 'react';
import './Start.scss';

class Start extends Component {
  constructor() {
    super()

  }
  render() {
    return (

  }
}

export default Start;