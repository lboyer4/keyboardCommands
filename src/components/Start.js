import React, { Component } from 'react';
import '../SCSS/_Start.scss';

class Start extends Component {
  constructor() {
    super()

  }

  handleSubmit = (e) => {
    e.preventDefault()
    let selectedLevel = e.target.parentElement.querySelector("input[type=radio]:checked").value
    this.props.setLevel(selectedLevel)
  }

  render() {
    return (
    <section className="start-holder">
    	<form>
      <p> Choose your level of difficulty to begin: </p>
        <input type="radio" id="Beginner-level" value="Beginner" name="level" />
          <label htmlFor="Beginner-level">Beginner</label>
        <input type="radio" id="Intermediate-level" value="Intermediate" name="level"/>
          <label htmlFor="Intermediate-level">Intermediate</label>
        <input type="radio" id="Advanced-level" value="Advanced" name="level"/>
          <label htmlFor="Advanced-level">Advanced</label>
        <button className="choose-level-btn" onClick={this.handleSubmit}>Choose Your Difficulty</button>
    	</form>
    </section>
    )
  }
}

export default Start;