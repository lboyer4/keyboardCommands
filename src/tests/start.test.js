import React from 'react';
import ReactDOM from 'react-dom';
import Start from '../components/Start';
import { shallow } from 'enzyme'; 

describe("Start", () => {
	let wrapper;

	const mocksetWrongCards = jest.fn();
	const mocksetLevel = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
    			<Start 
    				setLevel={mocksetLevel}
            setWrongCards={mocksetWrongCards}
            incorrectCards={[]}
            />
		)
	});

	it('should match snap shot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	 // it('should clear all when button is pushed', () => {
  //   wrapper.find('.choose-level-btn').simulate('click', {target: {}});
  //   expect(mocksetLevel).toBeCalled();
  // });

	 it('set the wrong cards when clicked', () => {
	 	wrapper.find('.wrong-cards-btn').simulate('click');
	 	expect(mocksetWrongCards).toBeCalled();
	 })
});