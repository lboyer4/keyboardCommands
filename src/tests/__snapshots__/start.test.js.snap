// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start should match snap shot 1`] = `
<section
  className="start-holder"
>
  <form>
    <p>
       To Begin, choose your level of diffuclty, or practice cards you've previously gotten incorrect
    </p>
    <input
      defaultChecked={true}
      id="Beginner-level"
      name="level"
      type="radio"
      value="Beginner"
    />
    <label
      htmlFor="Beginner-level"
    >
      Beginner
    </label>
    <input
      id="Intermediate-level"
      name="level"
      type="radio"
      value="Intermediate"
    />
    <label
      htmlFor="Intermediate-level"
    >
      Intermediate
    </label>
    <input
      id="Advanced-level"
      name="level"
      type="radio"
      value="Advanced"
    />
    <label
      htmlFor="Advanced-level"
    >
      Advanced
    </label>
    <div
      className="btn-container"
    >
      <button
        className="choose-level-btn"
        onClick={[Function]}
      >
        Begin Game
      </button>
      <button
        className="wrong-cards-btn"
        onClick={[Function]}
      >
        Practice Cards
      </button>
    </div>
  </form>
</section>
`;
